src/driver-steelseries.c

```diff
@@ -161,12 +185,28 @@ steelseries_probe(struct ratbag_device *device)
 		}
 
 		ratbag_profile_for_each_button(profile, button) {
+			const enum ratbag_button_type *types;
+			const struct ratbag_button_action *actions;
+
 			ratbag_button_enable_action_type(button, RATBAG_BUTTON_ACTION_TYPE_BUTTON);
 			ratbag_button_enable_action_type(button, RATBAG_BUTTON_ACTION_TYPE_SPECIAL);
 			ratbag_button_enable_action_type(button, RATBAG_BUTTON_ACTION_TYPE_KEY);
 
-			button->type = button_types[button->index];
-			ratbag_button_set_action(button, &button_actions[button->index]);
+			if (button_count <= 6) {
+				types = button_types_6;
+				actions = button_actions_6;
+			} else if (button_count == 8) {
+				types = button_types_8;
+				actions = button_actions_8;
+			} else {
+				log_debug(device->ratbag,
+					  "Device with %d buttons not supported\n",
+					  button_count);
+				return -ENOTSUP;
+			}
```

Comment:
Does it give us a lot of benefit to have a `_6` and a `_8` array here? I think it might be better to just build the arrays before the `for_each_button` loop, in the form of:
```
enum ratbag_button_type buttons[8] = {UNKNOWN};
buttons[0] = ...
buttons[1] = ...
if (nbuttons == 6)
  buttons[5] = ...
else
  buttons[5] = ...
```

This is just a nitpick and personal preference, so I'm not going to block on it. Would be good to check out how the code looks like with an approach like that.

Reply:
Thanks. The code is clearer that way. I separated the code to set the button defaults out to a function button_defaults_for_layout.